esda/tests/test_getisord.py

```diff
@@ -69,9 +69,14 @@ def test_G_star_Local_Binary(self, w):
         np.testing.assert_allclose(lg.p_sim[0], 0.102, rtol=RTOL, atol=ATOL)
 
     @parametrize_w
+    @pytest.mark.xfail(
+        reason="Intermittently does not warn for W param; reason unknown; see gh#331"
+    )
     def test_G_star_Row_Standardized(self, w):
-        with pytest.warns(UserWarning, match="Gi\\* requested, but"):
+        with pytest.warns(UserWarning, match="Gi\\* requested, but") as record:
             lg = getisord.G_Local(self.y, w, transform="R", star=True, seed=10)
+            if not record:
+                pytest.fail("Expected a warning!")
```

Comment:
Why do you do this?

Reply:
As a fail safe so it fails **exactly** when no warnings are emitted. I could not figure out how to add that granularity to `parametrize_w` without it affecting other tests. Maybe I'm missing something easy? Any thoughts?

I couldn't determine the proper `raises` argument for `pytest.mark.xfail` and I think having only a general failure is too vague.

Reply:
But this makes no difference, no? You are just doing explicitly what pytest would do here anyway. I probably don't follow here...

We can also split this into two tests, one checking the warning and the other the values.

Reply:
> I probably don't follow here...

I didn't explain well. What I mean it that I could not find the proper `pytest` error type to include in the [`xfail(raise=)` parameter](https://docs.pytest.org/en/stable/how-to/skipping.html#raises-parameter) - the error raised when no warnings are emitted when in `pytest.warns` context.

> We can also split this into two tests, one checking the warning and the other the values.

This may be the way to go if there isn't a more elegant solution that I am overlooking.

Reply:
I'd probably split it.